cedar-policy-core/src/ast/name.rs

```diff
@@ -89,6 +89,28 @@ impl Name {
     pub fn namespace(&self) -> String {
         self.path.iter().join("::")
     }
+
+    /// Prefix the name with a optional namespace
+    /// e.g., prefix `A::B`` with `Some(C)` produces `C::A::B`
+    /// prefix `A::B` with `None` yields itself
+    pub fn prefix_namespace(&self, namespace: Option<Name>) -> Name {
+        match namespace {
+            Some(namespace) => Self::new(
+                self.basename().clone(),
+                namespace
+                    .namespace_components()
+                    .chain(std::iter::once(namespace.basename()))
+                    .chain(self.namespace_components())
+                    .cloned(),
+            ),
+            None => self.clone(),
+        }
+    }
+
+    /// Test if a `Name` is an `Id`
+    pub fn is_singleton(&self) -> bool {
```

Comment:
```suggestion
    pub fn is_id(&self) -> bool {
```

Reply:
Or maybe `empty_namespace`

Reply:
Good suggestion. Maybe `is_namespace_empty`?

Reply:
On a second thought, `namespace` is not well defined for a name as it could be part of a namespace.

Reply:
Settled down on `is_unqualified`.